src/plugins/expressions/common/expression_types/specs/datatable.ts

```diff
@@ -23,6 +23,13 @@ import { ExpressionTypeDefinition } from '../types';
 import { PointSeries, PointSeriesColumn } from './pointseries';
 import { ExpressionValueRender } from './render';
 
+type State = string | number | boolean | null | undefined | SerializableState;
+
+/** @internal **/
+export interface SerializableState {
+  [key: string]: State | State[];
```

Comment:
nit: `Record<string, State | State[]>`

Reply:
actualy this will result in error: `error TS2456: Type alias 'State' circularly references itself.`